src/services/authentication/hooks/index.js

```diff
@@ -1,46 +1,97 @@
-const injectUserId = async (context) => {
-	const { systemId, schoolId, strategy } = context.data;
+const { TooManyRequests } = require('@feathersjs/errors');
 
-	if (schoolId) {
-		await context.app.service('schools').get(schoolId).then((school) => {
-			if (strategy === 'ldap') {
-				context.data.username = `${school.ldapSchoolIdentifier}/${context.data.username}`;
-			}
-		});
+const updateUsernameForLDAP = async (context) => {
+	const { schoolId, strategy } = context.data;
+
+	if (strategy !== 'jwt') {
+		if (schoolId) {
+			await context.app.service('schools').get(schoolId).then((school) => {
+				if (strategy === 'ldap') {
+					context.data.username = `${school.ldapSchoolIdentifier}/${context.data.username}`;
+				}
+			});
+		}
 	}
+	return context;
+};
 
-	return context.app.service('/accounts').find({
-		query: {
-			username: context.data.username,
-			systemId,
-		},
-		paginate: false,
-	}).then(async ([account]) => {
+const bruteForceCheck = async (context) => {
+	const { systemId, strategy } = context.data;
+
+	if (strategy !== 'jwt') {
+		const [account] = await context.app.service('/accounts').find({
+			query: {
+				username: context.data.username,
+				systemId,
+			},
+			paginate: false,
+		});
+
+		// if account doesn't exist we can not update (e.g. iserv, moodle)
 		if (account) {
-			context.params.payload = {};
-			context.params.payload.accountId = account._id;
-			if (account.userId) {
-				context.params.payload.userId = account.userId;
-			}
-			if (account.systemId) {
-				context.params.payload.systemId = account.systemId;
+			if (account.lastTriedLogin) {
+				const allowedTimeDifference = process.env.LOGIN_BLOCK_TIME || 15;
+				const timeDifference = (Date.now() - account.lastTriedLogin) / 1000;
+				if (timeDifference < allowedTimeDifference) {
+					throw new TooManyRequests(
+						'Brute Force Prevention!', {
+							timeToWait: allowedTimeDifference - Math.ceil(timeDifference),
+						},
+					);
+				}
 			}
-		} else if (['moodle', 'iserv'].includes(strategy)) {
-			const accountParameters = {
+			// set current time to last tried login
+			await context.app.service('/accounts').patch(account._id, { lastTriedLogin: Date.now() });
+		}
+	}
+	return context;
+};
+
+// Invalid Login will not call this function
```

Comment:
Shouldnt that be Valid login will not call this?

Reply:
it will reset the last login to 1970 if it is authenticated ... necessary for tests

Reply:
why? Can't you reset it in the test in question?

Reply:
Anyway it is not nice if you want to log in at your computer and tablet kind of the same time and then you get blocked for no reason

Reply:
only if you fail to login, right?

Reply:
right. it only blocks if the login fails